### PR TITLE
[Feature: #163554724] Allow a user the ability to request to be a mentor

### PR DESCRIPTION
**Pull request by rajeman**
#### What does this PR do?
- This PR ensures a user can request for mentorship

#### Description of Task to be completed?
- [x] create route for allowing a user to request mentorship
- [x] ensure a user with a pending or approved request cannot successfully make another request

#### How should this be manually tested?
-  goto `POST user/request` to make a mentorship request

#### What are the relevant pivotal tracker stories?
- [#163554724](https://www.pivotaltracker.com/n/projects/2229853/stories/163554724)

### CODE REVIEW

File: middlewares/requestValidations.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          [or]: ['pending', 'approved']
+        }
+      }
+    });
+    if (!currentRequest) {
+      return next();
+    }
+    const { status } = currentRequest;
+    const error = new Error(
+      status === 'pending' ? 'You already requested for mentorship'
```
Comment (oreoluwade) [inline middlewares/requestValidations.js]: Could you change this to `You already requested to be a mentor`?

File: controllers/request.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import models from '../models';
+
+const { Request } = models;
+
+export default {
+  async requestMentorship(req, res) {
```
Comment (oreoluwade) [inline controllers/request.js]: This might sound like asking for a mentor. Could you change to `requestToBeAMentor`? or something related?

**Review by oreoluwade — CHANGES_REQUESTED** [review]

File: controllers/request.js (hunk 2 above)
Comment (rajeman) [reply controllers/request.js]: I have done that

**Review by oreoluwade — APPROVED** [review]
Warp speed. Good job.

**Comment by rajeman** [comment]
Thanks

**Pull request merged**